pkg/account/deployer/client.go

```diff
@@ -0,0 +1,268 @@
+	if err = d.handleClientResponseError(resp, err, "unable to get policy with name: "+policy.Name); err != nil {
+		return "", err
+	}
+
+	existingPolicies := result.GetPolicies()
+
+	if len(existingPolicies) == 0 {
+		log.Debug("No policy with name %q found. Creating a new one", policy.Name)
+		var createdPolicy *accountmanagement.LevelPolicyDto
+		createdPolicy, resp, err = d.client.PolicyManagementAPI.CreateLevelPolicy(ctx, policyLevel, policyLevelId).CreateOrUpdateLevelPolicyRequestDto(policy).Execute()
+		if err = d.handleClientResponseError(resp, err, "unable to create policy with name: "+policy.Name); err != nil {
+			return "", err
+		}
+		return createdPolicy.GetUuid(), nil
+	}
+
+	if len(existingPolicies) > 1 { // shouldn't happen
+		log.Warn("Found multiple policies with name %q. Updating policy with UUID %q", policy.Name, existingPolicies[0].GetUuid())
+	}
+
+	log.Debug("Trying to update existing policy with name %q and UUID %q", policy.Name, existingPolicies[0].GetUuid())
+	_, resp, err = d.client.PolicyManagementAPI.UpdateLevelPolicy(ctx, policyLevel, policyLevelId, existingPolicies[0].GetUuid()).CreateOrUpdateLevelPolicyRequestDto(policy).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to update policy with name: "+policy.Name); err != nil {
+		return "", err
+	}
+	return existingPolicies[0].GetUuid(), nil
+}
+
+func (d *accountManagementClient) upsertGroup(ctx context.Context, groupId string, group Group) (remoteId, error) {
+	if groupId != "" {
+		log.Debug("Trying to update policy with origin object ID (UUID) %q", groupId)
+		resp, err := d.client.GroupManagementAPI.EditGroup(ctx, d.accountInfo.AccountUUID, groupId).PutGroupDto(group).Execute()
+
+		if err = d.handleClientResponseError(resp, err, "unable to update policy with UUID: "+groupId); err != nil {
+			return "", err
+		}
+		return groupId, nil
+	}
+
+	result, resp, err := d.client.GroupManagementAPI.GetGroups(ctx, d.accountInfo.AccountUUID).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to get groups with name: "+group.Name); err != nil {
+		return "", err
+	}
+
+	// find groups with matching name
+	var existingGroups []accountmanagement.GetGroupDto
+	for _, g := range result.GetItems() {
+		if g.GetName() == group.Name {
+			existingGroups = append(existingGroups, g)
+		}
+	}
+
+	if len(existingGroups) == 0 {
+		var createdGroups []accountmanagement.GetGroupDto
+		createdGroups, resp, err = d.client.GroupManagementAPI.CreateGroups(ctx, d.accountInfo.AccountUUID).PutGroupDto([]accountmanagement.PutGroupDto{group}).Execute()
+		if err = d.handleClientResponseError(resp, err, "unable to create group with name: "+group.Name); err != nil {
+			return "", err
+		}
+		if len(createdGroups) < 1 {
+			return "", fmt.Errorf("unable to get UUID of created group with name: %s", group.Name)
+		}
+		return createdGroups[0].GetUuid(), nil
+	}
+
+	if len(existingGroups) > 1 {
+		log.Warn("Updating multiple policies with name %s", group.Name)
```

Comment:
I think this is leftover wording from different logic? Like policies you're only updating the first. And I do think group names can't be duplicated right so this is also a "should never happen" case, right?

Reply:
yep